os/drivers/syslog/ramlog.c

```diff
@@ -678,6 +678,24 @@ int ramlog_consoleinit(void)
 {
 	FAR struct ramlog_dev_s *priv = &g_sysdev;
 
+	/* Initialize the non-zero values in the RAM logging device structure */
+
+	if ((priv->rl_exclsem.flags & FLAGS_INITIALIZED) == 0) {
+		sem_init(&priv->rl_exclsem, 0, 1);
+	}
+#ifndef CONFIG_RAMLOG_NONBLOCKING
+	if ((priv->rl_waitsem.flags & FLAGS_INITIALIZED) == 0) {
+
+		sem_init(&priv->rl_waitsem, 0, 0);
+
+		/*
+		 * The rl_waitsem semaphore is used for signaling and, hence,
+		 * should not have priority inheritance enabled.
+		 */
+		sem_setprotocol(&priv->rl_waitsem, SEM_PRIO_NONE);
```

Comment:
In original codes, there is no set of protocol type. Is this really indeed? Then, please split the commits for this.

Reply:
Don't you need this?

Reply:
It's needed because this semaphore is used for signaling. I updated with another commit.